src/Symfony/Component/VarDumper/Caster/Caster.php

```diff
@@ -65,11 +65,20 @@ public static function castObject($obj, $class, $hasDebugInfo = false)
         }
 
         if ($a) {
+            static $publicProperties = array();
+
             $i = 0;
             $prefixedKeys = array();
             foreach ($a as $k => $v) {
-                if (isset($k[0]) && "\0" !== $k[0] && !property_exists($class, $k)) {
```

Comment:
as shown by profiles in linked issue, this is a hot code path and `property_exists()` calling should be prevented